### DISCUSSION TITLE
Only 1 type of file in storage

### DISCUSSION THREAD

**Discussion by Arsh1a** [supabase/supabase#2001] [Questions] [answered]
How can I only let 1 type of file be uploaded into storage?

**Comment by ashleyconnor** [comment, marked as answer]
It's in one of the example policies:

> storage."extension"(name) = 'jpg'

Full example:

```sql
CREATE POLICY "policy_name"
ON storage.objects FOR {operation} {USING | WITH CHECK} (
  -- restrict bucket
  bucket_id = {bucket_name}
  -- allow access to only jpg file
  AND storage."extension"(name) = 'jpg'
  -- in the public folder
  AND LOWER((storage.foldername(name))[1]) = 'public'
  -- to anonymous users
  AND auth.role() = 'anon'
);
```

**Reply by sarkarology** [reply]
How to make it two file types, say jpg and png?

**Reply by GaryAustin1** [reply]
Without testing it probably AND (storage."extension"(name) = 'jpg' OR storage."extention"(name) = 'png')
If you had more than that then an array or list lookup might be way to go (ANY, IN)

**Reply by sarkarology** [reply]
Getting syntax error with:

(storage.extension(name) = IN('jpg','png'))

**Reply by sarkarology** [reply]
Sorry, my bad. Got it working with

(storage.extension(name) IN('jpg','png'))

**Reply by ammarfaris** [reply]
storage."extension"(name) = 'jpg' **does not seem to be robust enough**

1) if the file name consists of dots in the name, for example my.picture.jpg, 

`select storage."extension"(name)
from storage.objects;`

will return "picture" instead of "jpg"

2) on the client side, the file name can always be changed to a random extension, for example an svg file can be changed to .png on the client, and this by pass the checking as postgres will see the extension as "png" as opposed to the real svg extension/ file type.

is there another way to do this? perhaps targeting the type column [UPDATE: okay just realized that the type column is also set on the client]

![Screenshot 2022-08-09 at 11 37 23 AM](https://user-images.githubusercontent.com/1182205/183558741-6c2d294f-e3fd-4e27-9047-3b49e43239cf.png)

**Reply by ammarfaris** [reply]
another thing is, should we be worried about potential malicious svg files ? i see discourse.org convert svg files into png and Google profile image does not allow SVG files ? thank you.
